### PR TITLE
Implement workpiece entity

### PR DESCRIPTION
**Pull request by janosbrodbeck**
This adds the workpiece entity. Should be compatible with all current frontend versions.

* Add workpiece entity, which belongs to a project. Migration clears nearly all data again (sorry, necessary)
* For now, it is only possible to have a 1:1 relation with a project, although it is possible to support multiple workpieces per project later. But that completely changes the current workflow, as each workpiece would very likely have its own welding points.
* It is not possible to delete a workpiece!
* When a project is created, a workpiece is added to the project. My initial plan was that it is possible to have the option to optionally provide data for the workpiece as well with the upload of a project. But form data cannot have JSON data as well, it is not allowed by HTTP. So an empty workpiece is added to the project :( Either we have to send two requests when creating a project (meh), or just keep it empty and the user has to do it in the project in later stage (meh meh)
* Create project endpoint removed, as it was only a leftover option and I don't want to support this endpoint anymore. We create projects by uploading with a file
* `Upload` endpoint of project now also returns the workpiece! Same for a get request on the project with an ID. It returns the same as the upload endpoint now, so project + welding points + workpiece.
* The returned workpiece is always WITHOUT the model file. The model file has to be requested explicitly with a special endpoint of workpiece. Why? I have no idea how big model files are, and I don't want to attach a maybe multiple MB file to every project get request
* Endpoint tests for the workpiece. Skipped the other tests due to time reasons. For now the endpoints are IMHO the most important ones.

### CODE REVIEW

File: src/app/api/api_v1/endpoints/project.py [hunk 1]
```diff
@@ -36,9 +37,16 @@ async def read_project(
...
-    if not result:
+    project_obj = await project.get_by_id(db=db, id=project_id)
+    if not project_obj:
         raise HTTPException(status_code=404, detail="Project not found")
+
+    welding_points = await welding_point.get_multi_by_project_id(db=db, project_id=project_obj.id)
+    workpiece_obj = None
+    if project_obj.workpiece is not None:
+        workpiece_obj = await crud_workpiece.get_by_id(db=db, id=project_obj.workpiece.id
+                                                   )
```
Comment (dolevi-tub) [inline src/app/api/api_v1/endpoints/project.py:48]:
```suggestion
        workpiece_obj = await crud_workpiece.get_by_id(db=db, id=project_obj.workpiece.id)
```

File: src/app/api/api_v1/endpoints/project.py [hunk 2]
```diff
@@ -85,7 +93,7 @@ async def upload_project(
         db: AsyncSession = Depends(deps.get_async_db),
         name: str,
         description: Optional[str] = None,
-        file: UploadFile
+        file: UploadFile,
```
Comment (dolevi-tub) [inline src/app/api/api_v1/endpoints/project.py:96]:
```suggestion
        file: UploadFile
```

File: src/app/models/workpiece.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from sqlalchemy import Column, Integer, Float, String, ForeignKey
+from sqlalchemy.orm import relationship
+from app.db.base_class import Base
+
+
+class Workpiece(Base):
+    id = Column(Integer, primary_key=True, index=True)
+    project_id = Column(Integer, ForeignKey("project.id"))
```
Comment (dolevi-tub) [inline src/app/models/workpiece.py:8]:
```suggestion
    project_id = Column(Integer, ForeignKey("project.id"), nullable=False)
```

File: src/app/schemas/workpiece.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+                "position_y": 3.574,
+                "position_z": 4.154,
+            }
+        }
+
+
+class WorkpieceCreate(WorkpieceBase):
+    id: Optional[int]
+    project_id: int
+    model_file: Optional[str]
```
Comment (dolevi-tub) [inline src/app/schemas/workpiece.py:29]:
```suggestion
class WorkpieceCreate(WorkpieceBase):
    id: Optional[int]
    model_file: Optional[str]
```
The type of this property doesn't change from its parent class

**Review by dolevi-tub — CHANGES_REQUESTED** [review]
Had some minor remarks and like the decisions you made before implementing :+1:

**Comment by janosbrodbeck** [comment]
Addressed the change requests

**Pull request merged**